KvNGCzA/Hello-Books-BE
# 166575163 Borrow a book endpoint

#### What does this PR do?
It allows users to borrow a book endpoint

#### Description of Task to be completed?
- create borrow a book validator
- create a route to borrow a book
- add authorization and authentication
- create a borrow book controller
- create test cases
- add documentation

#### How should this be manually tested?
- Use **git pull** to pull branch to local branch
- Use **npm install** to download all necessary packages
- Update default config file to use environment variables of the project
- Ensure your database is created or you can use **sequelize db:create** if database does not exist
- Ensure that there is **no id** defined in any models
- Use **npm run db:roll** to create table in database
- Use **npm run test**  to test
- Go your POSTMAN and type **http://localhost:${PORT}/api/v1/borrow**

#### Any background context you want to provide?
NA

#### What are the relevant pivotal tracker stories?
[#166575163](https://www.pivotaltracker.com/story/show/166575163)

#### Screenshots (if appropriate)
![borrowedCapture](https://user-images.githubusercontent.com/7049701/59960902-a6d6ab80-94c7-11e9-9c7f-8dcbe146003f.PNG)
Sample of successful tests on borrowing a book route

![borrownotfound](https://user-images.githubusercontent.com/7049701/59632670-9f6b7700-9142-11e9-89c3-5152d08a1c0d.PNG)
Sample of not successful test on borrowing a book route (book not found)

![borrowalready](https://user-images.githubusercontent.com/7049701/59632733-ccb82500-9142-11e9-9f00-da3f2d41e9fb.PNG)

Sample of not successful test on borrowing a book route (book already borrowed)
#### Questions:
NA
